File: mobile/src/main/java/com/zacharytamas/often/models/managers/HabitManager.java
package com.zacharytamas.often.models.managers;

import android.content.Context;

import com.orm.query.Condition;
import com.orm.query.Select;
import com.zacharytamas.often.models.Habit;

import org.joda.time.DateTime;

import java.util.List;

/**
 * Created by zacharytamas on 10/25/15.
 */
public class HabitManager {

    private final Context mContext;

    public HabitManager(Context context) {
        mContext = context;
    }

    public List<Habit> getAvailableHabits() {
        return Select.from(Habit.class)
                .where(Condition.prop("available_at").lt(new DateTime().toDate().getTime())
                ).list();
    }

    public List<Habit> getAllHabits() {
        return Select.from(Habit.class).list();
    }
}

File: mobile/src/main/java/com/zacharytamas/often/models/managers/HabitManager.kt
package com.zacharytamas.often.models.managers

import android.content.Context
import com.orm.SugarRecord
import com.zacharytamas.often.models.Habit
import com.zacharytamas.often.utils.Data
import org.joda.time.DateTime

/**
 * A class for managing queries made for Habit objects.
 * Reminiscent of Django's Managers.
 */
class HabitManager(context: Context) {

    /**
     * Returns the Habits which are Available for completion at the current moment
     * and are explicitly due.
     */
//    fun getDueHabits() : RealmResults<Habit> {
//        return mRealm.where(Habit::class.java)
//                     .lessThan("availableAt", DateTime().toDate())
//                     .equalTo("required", true)
//                     .findAll()
//    }

    /**
     * Returns the Habits which are available for completion at the current moment.
     * This excludes Habits whose next occurrence is in the future.
     */
//    fun getAvailableHabits() : RealmResults<Habit> {
//        return mRealm.where(Habit::class.java)
//                     .lessThan("availableAt", DateTime().toDate())
//                     .equalTo("required", false)
//                     .findAll()
//    }

    fun completeHabit(habit: Habit) {
//        mRealm.beginTransaction()
//        Habits.completeHabit(habit)
//        mRealm.commitTransaction()
    }

    fun getAvailableHabits(): List<Habit> {
        // TODO This should actually query for habits that are available
        return SugarRecord.listAll(Habit::class.java)
    }
}